### DISCUSSION TITLE
Question regarding hard links

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#22434] [Q&A] [answered]
I've had osu!lazer on my pc for a couple years and i've imported around 6000 maps from stable to lazer in the past, and since hard links were introduced relatively recently, i'm wondering if i currently still have duplicated maps and if there's any procedure i need to do to properly make use of hard links and save space (i.e deleting all maps rand re-importing?), or is there nothing for me to worry about?

**Comment by Theighlin** [comment, marked as answer]
Hard links are created during the import process, so all the maps that you imported prior to the hard links support update are duplicated. I've read on another discussion of the possibility of retroactively applying hard links to beatmaps that are in both installation (and deleting the duplicate, to save space), but there's no information on when/if it will be done.

If you meet the hard links criteria (both installation are on the same drive, and it is NTFS if on windows), deleting and reimporting will work. Also, i think it's better if you restart the game between the 2 operations (lazer soft deletes maps until you exit the game).

**Reply by bdach** [reply]
All of the above is correct. Related: https://github.com/ppy/osu/discussions/21827
